modules/channel-slack/src/backend/client.ts

```diff
@@ -100,20 +109,45 @@ export class SlackClient {
     )
   }
 
-  private async _setupRealtime() {
+  private async _setupRealtime(port: number) {
     const discardedSubtypes = ['bot_message', 'message_deleted', 'message_changed']
-    this.rtm.on('message', async payload => {
-      debugIncoming(`Received real time payload %o`, payload)
-
-      if (!discardedSubtypes.includes(payload.subtype)) {
-        await this.sendEvent(payload, {
-          type: 'text',
-          text: _.find(_.at(payload, ['text', 'files.0.name', 'files.0.title']), x => x && x.length) || 'N/A'
-        })
-      }
-    })
 
-    return this.rtm.start()
+    if (this.rtm) {
+      this.rtm.on('message', async payload => {
+        debugIncoming(`Received real time payload %o`, payload)
+
+        if (!discardedSubtypes.includes(payload.subtype)) {
+          await this.sendEvent(payload, {
+            type: 'text',
+            text: _.find(_.at(payload, ['text', 'files.0.name', 'files.0.title']), x => x && x.length) || 'N/A'
+          })
+        }
+      })
+
+      return this.rtm.start()
+    } else {
+      this.events.on('message', async payload => {
+        debugIncoming(`Received real time payload %o`, payload)
+
+        if (!discardedSubtypes.includes(payload.subtype) && !payload.bot_id) {
+          await this.sendEvent(payload, {
+            type: 'text',
+            text: _.find(_.at(payload, ['text', 'files.0.name', 'files.0.title']), x => x && x.length) || 'N/A'
+          })
+        }
+      })
+
+      this.router.post(`/bots/${this.botId}/events-callback`, this.events.requestListener())
+      const publicPath = await this.router.getPublicPath()
+      this.logger.info(
+        `[${this.botId}] Events Endpoint URL: ${publicPath.replace('BOT_ID', this.botId)}/bots/${
+          this.botId
+        }/events-callback`
+      )
+
+      const server = await this.events.start(port || 0)
+      this.logger.info(`[${this.botId}] Slack events listening at : ${(<AddressInfo>server.address()).port}`)
```

Comment:
You are already registering a requestListener to handle incoming events. Slack provides 3 different ways to consume events, you only need one of them